drivers/counter/Kconfig.stm32_rtc

```diff
@@ -8,11 +8,11 @@
 menuconfig COUNTER_RTC_STM32
 	bool "STM32 Counter RTC driver"
 	depends on SOC_FAMILY_STM32
+	depends on !MINIMAL_LIBC
 	select USE_STM32_LL_RTC
 	select USE_STM32_LL_PWR
 	select USE_STM32_LL_RCC
 	select USE_STM32_LL_EXTI
-	select NEWLIB_LIBC
```

Comment:
I hope removing these `select`s will not cause too many issues to users..
Including these `select` changes would discourage from merging during freeze even though this PR includes a bugfix for another issue.
Would it make sense to separate the `select` changes commit into a separate PR for after 2.0?
(So just leaving this PR fixing only the bug, and not disturbing other options behavior, including the STM thingy, until after the release)
(The change of `depends on NEWLIB_LIBC` => `depends on !MINIMAL_LIBC` should be safe for applying now).

Reply:
>  I hope removing these selects will not cause too many issues to users

I hope the same  but that is the price we need to pay for using choice statement on the other hand the use of `select` is discouraged. 

> Including these select changes would discourage from merging during freeze even though this PR includes a bugfix for another issue.

Agree on this and think it is best to move this PR to 2.1

> Would it make sense to separate the select changes commit into a separate PR for after 2.0?

Short answer no, problem is that select does not work on choice symbols

```
warning: the choice symbol NEWLIB_LIBC (defined at lib/libc/Kconfig:21) is selected by the following symbols, but select/imply has no effect on choice symbols
 - CBOR_FLOATING_POINT (defined at modules/Kconfig.tinycbor:38)
 - CBOR_HALF_FLOAT_TYPE (defined at modules/Kconfig.tinycbor:44)
 - IEEE802154_KW41Z (defined at drivers/ieee802154/Kconfig.kw41z:9)
 - RTC_STM32 (defined at drivers/rtc/Kconfig.stm32_rtc:9)
 - COUNTER_RTC_STM32 (defined at drivers/counter/Kconfig.stm32_rtc:8)
 - UPDATEHUB (defined at lib/updatehub/Kconfig:7)
 - SIMPLELINK_HOST_DRIVER (defined at ext/hal/ti/simplelink/Kconfig:10)
```

Reply:
> select does not work on choice symbols

@vanwinkeljan Yep. I guess only keeping them as independent options would be possible to not change the selects.

If you would prefer to not go that route for a quick 2.0 fix, then I guess it would be nice to tag this PR and the bug with milestone set to 2.1 (or future)

(Note that separating the select->depend into its own commit does not really help if it is not possible to revert it independently or bisect between them)